### DISCUSSION TITLE
Workplane Size Control

### DISCUSSION THREAD

**Discussion by TheMakerTales** [hlorus/CAD_Sketcher#76] [Feature Requests]
The ability to set the size of a workplane. This could be both in the addon settings as a default size but would also be nice while it's first being created as a size value. Not sure if it should be percentage based or unit-based. Probably unit based.

Another option would be to have the workplane scale to have the same surface area as the face you click on, and the default workplane is the size of a workplane made from XYZ

**Comment by SupinePandora43** [comment]
How about to also center workplane on face you clicked on?

**Reply by TheMakerTales** [reply]
I think currently this is the default when you create a workplane from a mesh face no?

**Comment by MGlaab** [comment]
This is my first time with Github and making a feature request, so bear with me. I'm not sure if this is in the right place, but the topic is about workplanes.

I've watched a few YouTubes about workarounds for times when you want to reference geometry from another sketch in your current working sketch. The best workaround seems to be selecting the same workplane as the first sketch, solidifying the sketch geometry, then moving it into place. https://youtu.be/u9Vxu6aRlwE?si=JCsCcPJeh_YNepEV
This works until you edit the sketch, then the extruded mesh is moved back to the workplane. I tried applying the transforms but that still didn't work.

So, my request is the ability to move the workplane's position. We can already change the orientation just not the location, if you look in the Entities tab and click on the pencil icon. Though I'm not sure is moving the plane updates the geometry it seems like a good place to implement moving the workplane and the associated geometry. I hope that makes some sense.

So, the process would be making multiple sketches on multiple workplanes that all reference the same geometry, then being able to move those planes to their final locations and then use the solidify/extrude modifier.

**Reply by MGlaab** [reply]
I found a work around, you can align the workplane to center on the location of the 3D cursor. So, you can move the 3D cursor to your new location then align the workplane from the Entities tab and click on the pencil icon, Align Workplane to 3D Cursor.

**Reply by MGlaab** [reply]
you can move the plane but if the geometry references another sketch, the geometry doesn't move with the workplane.
